lib/transactions.js

```diff
@@ -235,29 +235,34 @@ TxController.prototype.list = function(req, res) {
       });
     });
   } else if(address) {
-    self.node.getAddressHistory(address, true, function(err, txinfos) {
+    self.node.getAddressHistoryCount(address, {}, function(err, historyCount) {
       if(err) {
         return common.handleErrors(err, res);
       }
 
-      var txs = txinfos.map(function(info) {
-        return info.tx;
-      }).filter(function(value, index, self) {
-        return self.indexOf(value) === index;
-      });
+      var options = {
+        from: page * pageLength,
+        to: Math.min((page + 1) * pageLength, historyCount)
```

Comment:
If you specify a "to" that is past the end it will do the min. It shouldn't be necessary to get the `historyCount`, see the "10 to 20 (out of range)" test: https://github.com/bitpay/bitcore-node/pull/223/files#diff-007c4e03d1b968d1804b4621a05c0885R523